clippy_lints/src/format_args.rs

```diff
@@ -559,6 +611,58 @@ impl<'tcx> FormatArgsExpr<'_, 'tcx> {
 
         false
     }
+
+    fn has_pointer_debug(&mut self, ty: Ty<'tcx>, depth: usize) -> bool {
+        let cx = self.cx;
+        let tcx = cx.tcx;
+        if depth > tcx.recursion_limit().0 {
```

Comment:
Nit, there is a method for this:

```suggestion
        if !tcx.recursion_limit().value_within_limit(depth) {
```